src/machines/fail_workflow_state_machine.rs

```diff
@@ -89,13 +90,13 @@ impl TryFrom<HistoryEvent> for FailWorkflowMachineEvents {
     type Error = WFMachinesError;
 
     fn try_from(e: HistoryEvent) -> Result<Self, Self::Error> {
-        Ok(match EventType::from_i32(e.event_type) {
-            Some(EventType::WorkflowExecutionFailed) => Self::WorkflowExecutionFailed,
+        Ok(match e.event_type() {
+            EventType::WorkflowExecutionFailed => Self::WorkflowExecutionFailed,
             _ => {
-                return Err(WFMachinesError::UnexpectedEvent(
-                    e,
-                    "Fail workflow machine does not handle this event",
-                ))
+                return Err(WFMachinesError::Nondeterminism(format!(
```

Comment:
Why is it attributed to non-determinism here?

Reply:
Because if the machine is processing an event it doesn't understand then that means user's WF code failed to produce a machine that *should've* matched it.